test/e2e/framework/util.go

```diff
@@ -0,0 +1,186 @@
+				return len(sa.Secrets) > 0, nil
+			case watch.Deleted, watch.Bookmark:
+				return false, nil
+			case watch.Error:
+				return true, apierrors.FromObject(e.Object)
+			default:
+				return true, fmt.Errorf("unexpected event type %v", t)
+			}
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return event.Object.(*corev1.ServiceAccount), nil
+}
+
+func WaitForObjectDeletion(ctx context.Context, dynamicClient dynamic.Interface, resource schema.GroupVersionResource, namespace, name string, uid types.UID) error {
+	fieldSelector := fields.OneTermEqualSelector("metadata.name", name).String()
+	lw := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			options.FieldSelector = fieldSelector
+			return dynamicClient.Resource(resource).Namespace(namespace).List(context.TODO(), options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.FieldSelector = fieldSelector
+			return dynamicClient.Resource(resource).Namespace(namespace).Watch(context.TODO(), options)
+		},
+	}
+	_, err := watchtools.UntilWithSync(
+		ctx,
+		lw,
+		&unstructured.Unstructured{},
+		func(store cache.Store) (bool, error) {
+			_, exists, err := store.Get(&metav1.ObjectMeta{Namespace: namespace, Name: name})
+			if err != nil {
+				return true, err
+			}
+			if !exists {
+				return true, nil
+			}
+			return false, nil
+		},
+		func(e watch.Event) (bool, error) {
+			switch t := e.Type; t {
+			case watch.Added, watch.Bookmark:
+				return false, nil
+			case watch.Modified:
+				// DeltaFIFO can return modified event on re-list if the object is recreated in the meantime
+				if e.Object.(metav1.Object).GetUID() != uid {
+					return true, nil
+				}
+				return false, nil
+			case watch.Deleted:
+				return true, nil
+			case watch.Error:
+				return true, apierrors.FromObject(e.Object)
+			default:
+				return true, fmt.Errorf("unexpected event type %v", t)
+			}
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO: Replace when Scylla cluster can configure a SA or use the default one.
+func SetupScyllaClusterSA(ctx context.Context, coreClient corev1client.CoreV1Interface, rbacClient rbacv1client.RbacV1Interface, namespace, scyllaClusterName string) error {
+	sa, err := coreClient.ServiceAccounts(namespace).Create(ctx, &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-member", scyllaClusterName),
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	_, err = rbacClient.RoleBindings(namespace).Create(ctx, &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: sa.Name,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				APIGroup:  corev1.GroupName,
+				Kind:      rbacv1.ServiceAccountKind,
+				Namespace: sa.Namespace,
+				Name:      sa.Name,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name:     "scylla-manager-cluster-member",
```

Comment:
scylla-manager?

Reply:
are you sure? this the rolebinding for the pod and it seemed to work so far

Reply:
It works by coincidence. CI script installs manager, and manager manifest deploys small scylla cluster with SA, role and role binding. This **scylla-manager-cluster**-member role is binded with SA used by the cluster under test. IMO we shouldn't depend here on role availability in CI, but instead create if from yaml in the test framework logic.

Reply:
Probably, the naming is a bit confusing in the current layout
manager-role?
https://github.com/scylladb/scylla-operator/blob/c5cfd75e383303909116aa9e368c8757dafb5232/config/operator/rbac/role.yaml#L7

Reply:
(also that clusterrole should have been prefixed with scylla or scylla-operator given it's in the global namespace)

Reply:
can you get me a link to which one you mean, the closest I see is `simple-cluster-member` (also good naming for non-namesapced obj) https://github.com/scylladb/scylla-operator/blob/a2b891ba3dff8c95a7bd8d627c665f04c02d4650/examples/generic/cluster.yaml#L13

Reply:
Naming isn't best I agree. Exactly this one `simple-cluister-member`. This role provides permission for sidecar to list services etc. 
Role name doesn't matter, but it must be binded with "<cluster.Spec.Name>-member"  SA.

Reply:
the issue is also coupling a deploy manifest to an example scylla - I suppose I'll have to parse it out with yq until we get manifests sorted out

Reply:
Or we can merge #451 first :D

Reply:
on the other hand you'll know you have done #451 correctly by being able to remove all the hacks from this PR. and also know your changes work because you'll have e2e coverage